### PR TITLE
Clear global resources while undeploy cluster

### PR DESCRIPTION
**Pull request by wuyingjun-lucky**
Signed-off-by: wuyingjun <wuyingjun_yewu@cmss.chinamobile.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Nice finding. I will take a look.

/assign

**Comment by wuyingjun-lucky** [comment]
> Nice finding. I will take a look.
> 
> /assign

ok, looking forward your suggestions and guidance

File: hack/undeploy-karmada.sh [hunk 1]
```diff
@@ -44,6 +44,25 @@ ETCD_HOST_IP=$(kubectl get pod -l app=etcd -n karmada-system -o jsonpath='{.item
...
+  for role in `kubectl  get clusterrole --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" | grep karmada-controller-manager |  awk '{print $1}' `
+  do
+    kubectl delete clusterrole $role --kubeconfig="${HOST_CLUSTER_KUBECONFIG}"
+  done
+}
+delete_cluster_roles
+
+# delete clusterrolebindings
+function delete_cluster_role_bindings() {
+  for rolebinding in `kubectl  get clusterrolebindings --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" | grep karmada-controller-manager |  awk '{print $1}' `
```
Comment (lonelyCZ) [inline hack/undeploy-karmada.sh:59]: Why are loops used here？Are there multiple `ClusterRole` resources that the name including `karmada-controller-manager`?

Comment (wuyingjun-lucky) [reply hack/undeploy-karmada.sh:59]: Yes 
 But I do not know when the clusterrole  created
image

Comment (lonelyCZ) [reply hack/undeploy-karmada.sh:59]: Right, I guess you registered `karmada-host` into control plane. And the namespace `karmada-cluster` in `karmada-host`  hasn't been removed.

Comment (wuyingjun-lucky) [reply hack/undeploy-karmada.sh:59]: Yes，and how should I modify the clear cluster role process

Comment (lonelyCZ) [reply hack/undeploy-karmada.sh:59]: I think we only remove all resources created by the control plane in `undeploy-karmada.sh`.

Comment (wuyingjun-lucky) [reply hack/undeploy-karmada.sh:59]: I got it

Comment (wuyingjun-lucky) [reply hack/undeploy-karmada.sh]: please help me review again

File: hack/undeploy-karmada.sh [hunk 2]
```diff
@@ -44,6 +44,13 @@ ETCD_HOST_IP=$(kubectl get pod -l app=etcd -n karmada-system -o jsonpath='{.item
 # clear all in namespace karmada-system
 kubectl delete ns karmada-system --kubeconfig="${HOST_CLUSTER_KUBECONFIG}"
 
+# clear global resources
```
Comment (lonelyCZ) [inline hack/undeploy-karmada.sh:47]: I guess this comment can be removed.

**Comment by lonelyCZ** [comment]
I have tested it that worked fine. Thanks @wuyingjun-lucky 

/lgtm

File: hack/undeploy-karmada.sh (hunk 2 above)
Comment (wuyingjun-lucky) [reply hack/undeploy-karmada.sh]: aggreed 
removing it looks better
done

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**